Make page margin configurable in CLI

uschek opened the pull request:
Hi,

I added some changes to the CLI of your nice application.

1. Added CLI parameters for top, left, bottom and right margin/border of pages to make the user override the default settings in the command line.
2. Changed the length values to require units (`--hHeight=20mm` instead of `--hHeight=20`). This is more consistent with the API where you have to pass units aswell. Also it makes it possible to use different units, e.g. `mm`, `cm` or `in`.
3. The default page margin is now `20mm` instead of `10mm`.

In my opinion, this makes CLI usage more comfortable. :) Let me know if you don't agree with any of these changes.

BlueHatbRit commented:
Hi @uschek, thanks for the contribution! Just wanted to let you know that I'm going to take a look at it today but the changes seem like a good idea, especially adding in the units.

BlueHatbRit commented:
@uschek thanks so much for the contribution again, the changes look great so far! I just had a question about the units of measurement.

Right now if you don't pass a unit with a particular measurement then it's ignored and the size you get out isn't quite what you might expect. I'm thinking we should check those parameters that are passed in and either throw an error asking the user to specify the unit or default to `mm` and return a message saying we've done so. What do you think?

I might have a bit of tunnel vision so it'd be good to get your feedback as a user before we add to the PR.

BlueHatbRit commented on bin/index.js:
```diff
@@ -17,15 +17,19 @@ const cli = meow(`
-        --hHeight  The height of the header section in mm
-        --footer   A HTML (.html) file to inject into the footer of the PDF
-        --fHeight  The height of the footer section in mm
-		--noEmoji  Disables emoji conversions
```
I didn't spot this, must have an editor configured incorrectly. Thanks for aligning them all.

uschek commented:
@BlueHatbRit Thanks for your feedback!

I didn't want to put too much logic in here as the `options.pdf`variable is [directly passed](https://github.com/BlueHatbRit/mdpdf/blob/2b0bd44b463d4030cbeb689d938f9cdd63c57734/index.js#L176) to [html-pdf](https://github.com/marcbachmann/node-html-pdf). I noticed it defaults to `px` when only a number is given. We should clarify the usage in the CLI help output and in the API docs by linking to the [options section of html-pdf](https://github.com/marcbachmann/node-html-pdf#options).

We should also be consistent with `html-pdf` by naming it `border` instead of `margin`. (I initially chose the latter because of the ability to include own CSS and `border` means something different there.)

I also think we should use `--dash-separated-options` instead of `--camelCaseOptions` as it's more common and you don't need any capital letters. `meow` already allows both ways, so it's just a change in the docs. :)

What do you think?

BlueHatbRit commented:
@uschek Ah I just found the default to `px`, I ran through a file and it seemed as if it ignored it completely but it's actually just very similar. I think we could add some clarity to the readme and help menu to cover things there.

I'm quite indifferent about margin vs border, I think you're right that border might make more sense as it's at least inline with html-pdf. I don't see it getting too confusing with anyone as the focus of the module is always the pdf, we don't really give any css guides. *So yes, I agree lets roll with border!* It's also probably a bit more friendly for non-css folk.

Yes you're right about dash separation, I'm quite honestly not sure why I went with camel case in the first place. I'll sort that on a separate PR before I ship your changes, just for a more atomic PR / commit history.

Thanks for engaging in some contribution, it's really appreciated. I hope you've found the module useful even before we ship these changes :)

uschek commented:
@BlueHatbRit Hope you had a great weekend.

I renamed `margin` to `border` and updated the docs accordingly. Now it's more consistent with `html-pdf`.

Let me know if there's anything to be changed before this can get merged.

I was looking for a tool as simple as possible and was about to write my own, but then I found this one. Thanks to you, too! :)

BlueHatbRit commented:
@uschek the changes look great, I'll get a release shipped asap (my new ISP should connect me sometime today). Thanks for adding in the hyphen separation as well, saves me a job.

Glad you're finding it useful!

BlueHatbRit commented:
I've made the release as version 1.4.0 on npm.

cc @uschek.

Thanks once again for your contribution, please feel free to send any new PR's or issues again! They're so appreciated and help make the project much better. 😄
